azure-mgmt-redis/src/main/java/com/microsoft/azure/management/redis/RedisCache.java

```diff
@@ -0,0 +1,497 @@
+        interface WithRedisConfiguration {
+            /**
+             * All Redis Settings. Few possible keys:
+             * rdb-backup-enabled, rdb-storage-connection-string, rdb-backup-frequency, maxmemory-delta, maxmemory-policy,
+             * notify-keyspace-events, maxmemory-samples, slowlog-log-slower-than, slowlog-max-len, list-max-ziplist-entries,
+             * list-max-ziplist-value, hash-max-ziplist-entries, hash-max-ziplist-value, set -max-intset-entries,
+             * zset-max-ziplist-entries, zset-max-ziplist-value etc.
+             *
+             * @param redisConfiguration configuration of Redis Cache as a map indexed by configuration name
+             * @return the next stage of Redis Cache update.
+             */
+            Update withRedisConfiguration(Map<String, String> redisConfiguration);
+
+            /**
+             * Specifies Redis Setting.
+             * rdb-backup-enabled, rdb-storage-connection-string, rdb-backup-frequency, maxmemory-delta, maxmemory-policy,
+             * notify-keyspace-events, maxmemory-samples, slowlog-log-slower-than, slowlog-max-len, list-max-ziplist-entries,
+             * list-max-ziplist-value, hash-max-ziplist-entries, hash-max-ziplist-value, set -max-intset-entries,
+             * zset-max-ziplist-entries, zset-max-ziplist-value etc.
+             *
+             * @param key   Redis configuration name.
+             * @param value Redis configuration value.
+             * @return the next stage of Redis Cache update.
+             */
+            Update withRedisConfiguration(String key, String value);
+
+            /**
+             * Cleans all the configuration settings being set on Redis Cache.
+             *
+             * @return the next stage of Redis Cache update.
+             */
+            Update withoutRedisConfiguration();
+
+            /**
+             * Removes specified Redis Cache configuration setting.
+             *
+             * @param key Redis configuration name.
+             * @return the next stage of Redis Cache update.
+             */
+            Update withoutRedisConfiguration(String key);
+        }
+    }
+
+    /**
+     * The template for a Redis Cache update operation, containing all the settings that can be modified.
+     */
+    interface Update extends
+            Appliable<RedisCache>,
+            Resource.UpdateWithTags<Update>,
+            UpdateStages.WithSku,
+            UpdateStages.WithNonSslPort,
+            UpdateStages.WithRedisConfiguration {
+
+        /**
+         * The full resource ID of a subnet in a virtual network to deploy the Redis Cache in.
+         * Example format: /subscriptions/{subid}/resourceGroups/{resourceGroupName}/Microsoft.{Network|ClassicNetwork}/VirtualNetworks/vnet1/subnets/subnet1
+         *
+         * @param subnetId the subnetId value to set.
+         * @return the next stage of Redis Cache update.
+         */
+        Update withSubnetId(String subnetId);
```

Comment:
we do not currently provide a way in the API to get a Subnet ID. So in cases where the code needs to point at a specific subnet reference, we let the user point at the parent network (as string resource ID ok if from a package that doesn't reference Networking), and the name of the subnet as string. For an example of that kind of reference, see `PrivateFrontend` in networking - there is `networkId()` and `subnetName()`, and then see `withExistingSubnet()` among its DefinitionStages.